### DISCUSSION TITLE
Modules extension: Allow to hide nav element

### DISCUSSION THREAD

**Discussion by raxon** [directus/directus#12417] [Feature Requests]
Hi, I have been working for some time with extensions for directus, and during creation I miss the possibility to hide the nav-module element in the extension module like on screen below: 

![image](https://user-images.githubusercontent.com/1338844/160298728-91a15e05-08e2-40d2-9413-52358265267b.png)

**Comment by benhaynes** [comment]
There is some system information in this bar, so I think the best option is to have it hide, with some visual indicator on how to temporarily expand it. It could also be programmatically expanded as needed. For certain context. The project/latency information could be moved to a tool tip on the logo.

**Comment by raxon** [comment]
Other way is add  like on screen Orange selection add margin from top and remove left nav but red top bar extend to full width.
Then functionality does not change but bar still exist in a part.


![CleanShot 2022-03-28 at 09 28 04@2x](https://user-images.githubusercontent.com/1338844/160348087-29b78ee3-3a8e-4692-8661-7c3dd052e16a.png)

**Reply by benhaynes** [reply]
Haha, I can't believe I didn't see this! Smart... that keeps the Project Info available. The only issue is that the page would no longer align with the header info, but that might not matter too much.

**Reply by raxon** [reply]
Solution: If we separate the navigation and sub-nav on the left side, maybe flexbox will solve the problem, and when we need a menu, flex-direction:row will make it look like normal, but when we disable flex-direction:col, the sub-nav will be the width of the whole page, and the other one will be the data from Container.

**Reply by benhaynes** [reply]
True — also, some of these Layouts/Modules may not be the typical form, where alignment matters. I'm thinking of things like the Map Layout, where there's no "left edge"  to the page.

**Reply by OzzyW80** [reply]
Funny you bring this up because I have been thinking about it last few days. My thought have been if it could be an option to click to select the tablet/mobile menu on desktop, this would be resolved. Imagine a button only visible on hover saying close sidebar. When user clicks the mobile menu is activated. I guess the content would have to be more aligned center than it is today. Not sure why it isn't already, is the space to the right of the forms saved for preview for later?

**Reply by OzzyW80** [reply]
https://user-images.githubusercontent.com/96046228/160603886-dbaa6d39-0b8f-4503-a7e8-01e4dc8f06f4.mov

**Reply by benhaynes** [reply]
Makes sense... re-use the mobile system we already have.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
